File: src/main/java/com/Khata/Khata/Controller/UserController.java
package com.Khata.Khata.Controller;

import com.Khata.Khata.Entity.User;
import com.Khata.Khata.Service.UserService;
import org.springframework.security.core.Authentication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/me")
public class UserController
{
    private final UserService userService;

    UserController(UserService userService)
    {
        this.userService = userService;
    }

    @GetMapping
    public User me(Authentication auth)
    {
        return userService.getUserById((Integer) auth.getPrincipal());
    }
}

File: classtracker/src/main/kotlin/com/classtrackeer/classtracker/Controller/UserController.kt
package com.classtrackeer.classtracker.Controller

import com.classtrackeer.classtracker.Model.User
import com.classtrackeer.classtracker.Service.UserService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1/user")
class UserController (@Autowired val userService : UserService){

    @PostMapping("/addUser")
    fun addUser(@RequestBody user : User) : ResponseEntity<User> {
        val savedUser : User = userService.addUser(user)
        return ResponseEntity(savedUser, HttpStatus.OK)
    }
}